app/services/catalog/update_order_item.rb

```diff
@@ -0,0 +1,55 @@
+module Catalog
+  class UpdateOrderItem
+    class ServiceInstanceWithoutExternalUrl < StandardError; end
+
+    def initialize(topic)
+      @payload = topic.payload
+      @message = topic.message
+    end
+
+    def process
+      Rails.logger.info("Processing service order topic message: #{@message} with payload: #{@payload}")
+
+      Rails.logger.info("Searching for OrderItem with a task_ref: #{@payload["task_id"]}")
+      order_item = find_order_item
+      Rails.logger.info("Found OrderItem: #{order_item.id}")
+
+      ManageIQ::API::Common::Request.with_request(order_item.context.transform_keys(&:to_sym)) do
+        order_item.update_message("info", "Task update message received with payload: #{@payload}")
+
+        if @payload["state"] == "completed"
```

Comment:
@eclarizio We are not doing anything with the order here, an order comprises of one or more order items.
When an order item changes state we have to change the order of the state.
You need to call this code
https://github.com/ManageIQ/catalog-api/blob/6123893880c14260bd9571cae9caf044bc1cedaa/app/models/order.rb#L19 which will move the order to completion.

Also the order_item has this code
https://github.com/ManageIQ/catalog-api/blob/6123893880c14260bd9571cae9caf044bc1cedaa/app/models/order_item.rb#L57

Which needs to be moved here.
Actually both the order_item and order functions need to be moved from app/models to here.